packages/maps/src/components/basic/GeoDistanceDropdown.js

```diff
@@ -49,74 +42,104 @@ class GeoDistanceDropdown extends Component {
 		if (props.autoLocation) {
 			this.getUserLocation();
 		}
+
+		let currentLocation = null;
+		let currentDistance = 0;
+
 		props.setQueryListener(props.componentId, props.onQueryChange, null);
-	}
+		props.addComponent(this.props.componentId);
+		this.setReact(props);
 
-	componentWillMount() {
-		this.props.addComponent(this.props.componentId);
-		this.setReact(this.props);
+		if (props.value) {
+			currentLocation = props.value.location;
+			const selected = props.data.find(
+				item => item.label === props.value.label,
+			);
 
-		if (this.props.selectedValue) {
-			this.setState({
-				currentLocation: this.props.selectedValue.location,
-			});
-			this.getCoordinates(this.props.selectedValue.location, () => {
-				const selected = this.props.data.find(
-					item => item.label === this.props.selectedValue.label,
-				);
-				this.setDistance(selected.distance);
-			});
-		} else if (this.props.defaultSelected) {
-			this.setState({
-				currentLocation: this.props.defaultSelected.location,
-			});
-			this.getCoordinates(this.props.defaultSelected.location, () => {
-				const selected = this.props.data.find(
-					item => item.label === this.props.defaultSelected.label,
-				);
-				this.setDistance(selected.distance);
-			});
+			currentDistance = selected.distance;
+		} else if (props.selectedValue) {
+			currentLocation = props.selectedValue.location;
+			const selected = props.data.find(
+				item => item.label === props.selectedValue.label,
+			);
+
+			currentDistance = selected.distance;
+		} else if (props.value) {
```

Comment:
this condition is same as `if` condition so it will never execute this block

Reply:
Thanks will fix this